core/src/main/java/com/linecorp/armeria/client/DefaultClientRequestContext.java

```diff
@@ -586,6 +586,16 @@ public boolean isTimedOut() {
         return responseTimeoutScheduler.isTimedOut();
     }
 
+    @Override
+    public void cancel() {
+        responseTimeoutScheduler.cancel();
```

Comment:
It seems like canceling the response timeout to me. Should we rename the `TimeoutScheduler`? 🤔 
/cc @ikhoon

Reply:
Agreed. How about making `timeoutNow()` take an Exception that could be used when `TimeoutTask` is invoked?

```java
// in TimeoutScheduler
void timeoutNow(Throwable cause) {
   ...
}
@Nullable
Throwable timeoutCause() { ... }

// in HttpResponseDecoder
Throwable cause = timeoutScheduler.timeoutCause();
if (cause == null) {
    cause = ResponseTimeoutException.get();
} 
...
```

Reply:
I think `cancel()` is clearer about the behavior we want. Having a `cancel()`, `isCancelled()` pair looks better. Also I don't think we need to pass the `cause` around, the `TimeoutTask` can determine it in one place. But I agree on renaming `TimeoutScheduler` to something more general 😄